extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/serde/JsonbSerde.java

```diff
@@ -0,0 +1,122 @@
+package io.quarkus.kafka.client.serde;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+
+/**
+ * A {@link Serde} that (de-)serializes JSON using JSON-B.
+ */
+public class JsonbSerde<T> implements Serde<T> {
+
+    private final Class<T> type;
+    private final Jsonb jsonb;
+
+    public JsonbSerde(Class<T> type) {
+        this(type, JsonbBuilder.create());
+    }
+
+    public JsonbSerde(Class<T> type, Jsonb jsonb) {
+        this.type = type;
+        this.jsonb = jsonb;
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+    }
+
+    @Override
+    public void close() {
+        try {
+            jsonb.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Serializer<T> serializer() {
+        return new JsonbSerializer<T>();
+    }
+
+    @Override
+    public Deserializer<T> deserializer() {
+        return new JsonbDeserializer<T>(type);
+    }
+
+    public static class JsonbDeserializer<U> implements Deserializer<U> {
```

Comment:
Should it be a separate class to the configuration file can refer to this one "more easily"?

Reply:
Yes, good idea. Reworked the commit accordingly.